plugins/device-injector/device-injector.go

```diff
@@ -104,6 +107,31 @@ func (p *plugin) CreateContainer(_ context.Context, pod *api.PodSandbox, contain
 		}
 	}
 
+	// inject CDI devices to container
+	cdiDevices, err = parseCDIDevices(container.Name, pod.Annotations)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if len(cdiDevices) == 0 {
```

Comment:
Does an early return for this case make sense? It would remove the need to have the injection code in an else block.

Reply:
We can't really, because the same function later might still inject mounts if those were annotated.

Reply:
OK. I see now. I think separate functions would be more readable, but I'm also happy for that to be a follow-up.

Reply:
Refactored to use separate functions, a common function for picking annotations, and to use an early return to omit a final else-block in the injection code paths.